### PR TITLE
Cxl.v9

### LINKED ISSUE #74: Support CXL.memcache in teeio-validator

### PR DESCRIPTION
**Pull request by mxu9**
Fix #74

### CODE REVIEW

File: teeio-validator/library/cxl_ide_test_lib/cxl_ide_common.c [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+  {"GetKey",      "1",                          CXL_MEM_IDE_TEST_CASE_GETKEY    },
+  {"Test",        CXL_IDE_TEST_CLASS_CASE_NAMES,CXL_MEM_IDE_TEST_CASE_TEST      },
+  {NULL,          NULL,                         CXL_MEM_IDE_TEST_CASE_NUM       }
+};
+
+
+ide_test_case_funcs_t m_cxl_ide_test_case_funcs[CXL_MEM_IDE_TEST_CASE_NUM][MAX_CXL_CASE_ID] = {
+  // Query
+  {
+    {NULL, NULL, NULL, false},
```
Comment (jyao1) [inline teeio-validator/library/cxl_ide_test_lib/cxl_ide_common.c:73]: what's this?

File: teeio-validator/library/cxl_ide_test_lib/test_config/test_config_default.c [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  ide_common_test_config_context_t *config_context = (ide_common_test_config_context_t *)test_context;
+  TEEIO_ASSERT(config_context->signature == CONFIG_CONTEXT_SIGNATURE);
+
+  cxl_ide_test_group_context_t *group_context = config_context->group_context;
+  TEEIO_ASSERT(group_context->signature == GROUP_CONTEXT_SIGNATURE);
+
+  CXL_PRIV_DATA* rp_cxl_data = &group_context->upper_port.cxl_data;
+  CXL_PRIV_DATA* ep_cxl_data = &group_context->lower_port.cxl_data;
+
+  // TODO
```
Comment (jyao1) [inline teeio-validator/library/cxl_ide_test_lib/test_config/test_config_default.c:41]: what is this TODO?

File: teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+      &tx_key_buffer, &kp_ack_status);
+  if (LIBSPDM_STATUS_IS_ERROR(status))
+  {
+    TEEIO_DEBUG((TEEIO_DEBUG_ERROR, "cxl_ide_km_key_prog TX failed with status=0x%08x\n", status));
+    TEEIO_ASSERT(false);
+    return false;
+  }
+  TEEIO_DEBUG((TEEIO_DEBUG_INFO, "key_prog TX - %02x\n", kp_ack_status));
+
+  // TODO
```
Comment (jyao1) [inline teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c:147]: what is this TODO?

File: teeio-validator/library/cxl_ide_test_lib/cxl_ide_common.c (hunk 1 above)
Comment (mxu9) [reply teeio-validator/library/cxl_ide_test_lib/cxl_ide_common.c]: Test cases for Query/KeyProg/KSetGo/KSetStop/GetKey have not been implemented. These are the placeholders in m_cxl_ide_test_case_funcs.
In the next version, above funcs will be added with dumb functions like below:
```
bool cxl_ide_test_query_1_setup(void *test_context)
{
  NOT_IMPLEMENTED(__func__);
  return false;
}

bool cxl_ide_test_query_1_run(void *test_context)
{
  NOT_IMPLEMENTED(__func__);
  return false;
}

bool cxl_ide_test_query_1_teardown(void *test_context)
{
  NOT_IMPLEMENTED(__func__);
  return false;
}
```

File: teeio-validator/library/cxl_ide_test_lib/test_config/test_config_default.c (hunk 2 above)
Comment (mxu9) [reply teeio-validator/library/cxl_ide_test_lib/test_config/test_config_default.c]: This TODO shall be deleted, together with other un-used lines in the codes.

File: teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c (hunk 3 above)
Comment (mxu9) [reply teeio-validator/library/cxl_ide_test_lib/test_case/ide_km_common.c]: This TODO shall be deleted, together with other un-used lines in the codes.

**Pull request merged**